src/bootstrap/src/core/download.rs

```diff
@@ -250,19 +257,113 @@ impl Config {
+                    .arg(commit)
+                    .output()
+                    .unwrap()
+                    .stdout
+                    .into_iter()
+                    .map(|c| c as char)
+                    .collect();
+                if build_date.is_empty() {
+                    eprintln!("NOTE: trying to download builds for {}", commit);
+                } else {
+                    eprintln!(
+                        "NOTE: trying to download builds for {}, but it is from {}",
+                        commit, build_date
+                    );
+                }
+            } else {
+                eprintln!("NOTE: you seem to have an outdated version of rust source");
+            }
+        };
+
+        let user: String = Command::new("git")
+            .arg("config")
+            .arg("user.name")
+            .output()
+            .expect("Failed to get git user.name")
+            .stdout
+            .into_iter()
+            .map(|c| c as char)
+            .collect();
+        let upstream = git::get_rust_lang_rust_remote(
+            &GitConfig { git_repository: "rust-lang/rust.git", nightly_branch: "" },
+            None,
+        );
+        match upstream {
+            Ok(upstream) => {
+                let log: String = Command::new("git")
+                    .arg("log")
+                    .arg("--pretty=short")
+                    .arg(format!("{}/master..HEAD", upstream))
+                    .output()
+                    .expect("Failed to get git log")
+                    .stdout
+                    .into_iter()
+                    .map(|c| c as char)
+                    .collect();
+                for s in log.split("\n") {
+                    if s.contains("Author:") {
+                        if !s.contains(&format!("Author: {}", user.replace("\n", ""))) {
+                            check_outdated_msg();
+                            return;
+                        }
+                    }
+                }
+            }
+            Err(e) => eprintln!("{}", e),
+        }
+        let last_commit: String = Command::new("git")
+            .arg("show")
+            .arg("-s")
+            .arg("--date=short")
+            .arg("--format=%ct")
+            .arg("origin/master")
+            .output()
+            .expect("Failed to get git log")
+            .stdout
+            .into_iter()
+            .map(|c| c as char)
+            .collect();
+        match SystemTime::now().duration_since(SystemTime::UNIX_EPOCH) {
+            Ok(n) => {
+                let replaced = last_commit.replace("\n", "");
+                let diff = n.as_secs() - replaced.parse::<u64>().unwrap();
+                if diff >= 165 * 24 * 60 * 60 {
+                    // origin/master is more than 165 days out of date, CI builds disappear in 168 days
+                    eprintln!(
+                        "NOTE: origin/master is {} days out of date, CI builds disappear in 168 days.",
```

Comment:
origin/master isn't necessarily the right thing. Use the same detection as above for finding the right name of the remote.